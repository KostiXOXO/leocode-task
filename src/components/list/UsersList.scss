@use '/src/styles/colors' as *;

.listContainer {
	&.ordered {
		list-style: decimal;
	}

	&.unordered {
		list-style: disc;
	}

	&.ordered,
	&.unordered {
		line-height: 1.5;
		li div {
			display: flex;
			gap: 0.7rem;
		}

		li::marker,
		li div .username {
			color: $clr-text-secondary;
			font-size: 0.8rem;
			display: flex;
			align-items: flex-end;
		}
	}
}
