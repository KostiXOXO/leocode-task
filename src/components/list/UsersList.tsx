import React from 'react';
import { DefaultListProps } from './common';
import './UsersList.scss';

const UsersList = ({ type = 'unordered', items }: DefaultListProps) => {
	const itemsListed = items?.map(({ name, username }, key) => {
		return (
			<li key={key}>
				<div>
					<span className="name">
						<strong>{name}</strong>
					</span>
					<span className="username">{`@${username}`}</span>
				</div>
			</li>
		);
	});

	return type === 'unordered' ? (
		<ul className="listContainer unordered">{itemsListed}</ul>
	) : (
		<ol className="listContainer ordered">{itemsListed}</ol>
	);
};

export default UsersList;
