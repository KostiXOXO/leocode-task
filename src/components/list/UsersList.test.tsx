import React from 'react';
import { cleanup, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import UsersList from './UsersList';

afterEach(cleanup);

const items = [
	{ name: 'Krystian', username: 'Kosti' },
	{ name: 'User1', username: 'Username1' },
];

it('Renders UserList with given name and username starting with @', () => {
	const { getByText } = render(<UsersList items={items} />);
	getByText('Krystian');
	getByText('@Kosti');

	getByText('User1');
	getByText('@Username1');
});

it('Renders with a className equal to the type', () => {
	render(<UsersList type="ordered" items={items} />);
	const listComponentOrdered = screen.getByTestId('listComponent');
	expect(listComponentOrdered).toHaveClass('ordered');

	cleanup();

	render(<UsersList type="unordered" items={items} />);
	const listComponentUnordered = screen.getByTestId('listComponent');
	expect(listComponentUnordered).toHaveClass('unordered');
});

it('While no type given - renders with a className equal to unordered', () => {
	render(<UsersList items={items} />);
	const listComponent = screen.getByTestId('listComponent');
	expect(listComponent).toHaveClass('unordered');
});

it('Component did not change', () => {
	const tree = renderer.create(<UsersList items={items} />).toJSON();
	expect(tree).toMatchSnapshot();
});
