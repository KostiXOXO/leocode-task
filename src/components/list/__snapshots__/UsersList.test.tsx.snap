// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component did not change 1`] = `
<ul
  className="listContainer unordered"
  data-testid="listComponent"
>
  <li>
    <div>
      <span
        className="name"
      >
        <strong>
          Krystian
        </strong>
      </span>
      <span
        className="username"
      >
        @Kosti
      </span>
    </div>
  </li>
  <li>
    <div>
      <span
        className="name"
      >
        <strong>
          User1
        </strong>
      </span>
      <span
        className="username"
      >
        @Username1
      </span>
    </div>
  </li>
</ul>
`;
